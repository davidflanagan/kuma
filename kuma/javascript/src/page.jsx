// @flow
import * as React from 'react';
import { useContext, useEffect } from 'react';
import { css } from '@emotion/core';

import Article from './article.jsx';
import DocumentProvider from './document-provider.jsx';
import GAProvider from './ga-provider.jsx';
import { gettext } from './l10n.js';
import LanguageMenu from './header/language-menu.jsx';
import Header from './header/header.jsx';
import TaskCompletionSurvey from './task-completion-survey.jsx';
import { navigateRenderComplete } from './perf.js';

import type { DocumentData } from './document-provider.jsx';
type DocumentProps = {
    document: DocumentData
};

// A media query that identifies screens narrower than a tablet
const NARROW = '@media (max-width: 749px)';

const styles = {
    // Titlebar styles
    titlebarContainer: css({
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        boxSizing: 'border-box',
        width: '100%',
        minHeight: 106,
        padding: '12px 24px',
        overflowX: 'scroll',
        backgroundColor: '#f5f9fa',
        borderBottom: 'solid 1px #dce3e5',
        borderTop: 'solid 1px #dce3e5',
        [NARROW]: {
            // Reduce titlebar size on narrow screens
            minHeight: 60,
            padding: '8px 16px'
        }
    }),
    titlebar: css({
        display: 'flex',
        flexDirection: 'row',
        width: '100%',
        maxWidth: 1352
    }),
    title: css({
        fontFamily:
            'x-locale-heading-primary, zillaslab, "Palatino", "Palatino Linotype", x-locale-heading-secondary, serif',
        fontSize: 45,
        fontWeight: 'bold',
        [NARROW]: {
            // Reduce the H1 size on narrow screens
            fontSize: 28
        }
    }),

    // Breadcrumbs styles
    breadcrumbsContainer: css({
        boxSizing: 'border-box',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
        minHeight: 40,
        padding: '4px 16px 4px 24px',
        borderBottom: 'solid 1px #dce3e5',
        backgroundColor: '#fff'
    }),
    breadcrumbsRow: css({
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        width: '100%',
        maxWidth: 1360
    }),
    breadcrumbs: css({
        flex: '1 1'
    }),
    crumb: css({
        display: 'inline',
        fontSize: 14,
        '&a': {
            color: '#3d7e9a'
        }
    }),
    chevron: css({
        fontFamily: 'zillaslab,serif',
        fontWeight: 'bold',
        fontSize: 18,
        margin: '0 8px'
    }),

    // Content styles: the sidebar and main article content
    contentLayout: css({
        display: 'grid',
        boxSizing: 'border-box',
        maxWidth: 1400,
        margin: '0 auto',
        gridTemplateColumns: '25% 75%',
        gridTemplateAreas: '"side main"',
        [NARROW]: {
            // If we're narrower than a tablet, put the sidebar at the
            // bottom and drop the toc line.
            gridTemplateColumns: '1fr',
            gridTemplateAreas: '"main" "side"'
        }
    }),
    sidebar: css({
        gridArea: 'side',
        boxSizing: 'border-box',
        width: '100%',
        overflowX: 'scroll',
        // Less padding on the right because the article area
        // has padding on the left, too.
        padding: '30px 12px 30px 24px',
        [NARROW]: {
            // Except that on small screens the sidebar is at the bottom and
            // so we need the same padding (but less of it) on both sides.
            padding: '15px 12px'
        }
    }),
    related: css({
        fontSize: 20
    })
};

function Titlebar({ document }: DocumentProps) {
    return (
        <div css={styles.titlebarContainer}>
            <div css={styles.titlebar}>
                <h1 css={styles.title}>{document.title}</h1>
            </div>
        </div>
    );
}

function Sidebar({ document }: DocumentProps) {
    return (
        <div className="quick-links" css={styles.sidebar}>
            <div css={styles.related} className="quick-links-head">
                {gettext('Related Topics')}
            </div>
            <div
                dangerouslySetInnerHTML={{
                    __html: document.quickLinksHTML
                }}
            />
        </div>
    );
}

function Chevron() {
    return <span css={styles.chevron}>â€º</span>;
}

function Breadcrumbs({ document }: DocumentProps) {
    // The <span> elements below aren't needed except that the stylesheets
    // are set up to expect them.
    return (
        <div css={styles.breadcrumbsContainer}>
            <div css={styles.breadcrumbsRow}>
                <nav css={styles.breadcrumbs} role="navigation">
                    <ol>
                        {document.parents.map(p => (
                            <li css={styles.crumb} key={p.url}>
                                <a href={p.url}>
                                    <span>{p.title}</span>
                                </a>
                                <Chevron />
                            </li>
                        ))}
                        <li css={styles.crumb}>
                            <span>{document.title}</span>
                        </li>
                    </ol>
                </nav>
                <LanguageMenu />
            </div>
        </div>
    );
}

function Content({ document }: DocumentProps) {
    return (
        <div css={styles.contentLayout}>
            <Article document={document} />
            <Sidebar document={document} />
        </div>
    );
}

export default function Page() {
    const document = useContext(DocumentProvider.context);
    const ga = useContext(GAProvider.context);

    /*
     * Register an effect that runs every time we see a new document URL.
     * The effect sends a Google Analytics timing event to record how long
     * it took from the start of the navigation until the new document is
     * rendered.
     *
     * Effects don't run during server-side-rendering, but that is fine
     * because we only want to send the GA event for client-side navigation.
     * Calling navigateRenderComplete() will have no effect if
     * navigateStart() was not previously called.
     */
    useEffect(() => {
        navigateRenderComplete(ga);
    }, [document && document.absoluteURL]);

    return (
        document && (
            <>
                <Header />
                <TaskCompletionSurvey />
                <Titlebar document={document} />
                <Breadcrumbs document={document} />
                <Content document={document} />
            </>
        )
    );
}
