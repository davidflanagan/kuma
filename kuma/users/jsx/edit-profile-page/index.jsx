import React from "react";
import ReactDOM from "react-dom";
import EditProfilePage from './edit-profile-page.jsx';

let container = document.getElementById('react');
let props = container.dataset;
let csrftoken = document.cookie
    .split(';')
    .filter(s => s.startsWith('csrftoken='))[0]
    .slice(10);

function render() {
    ReactDOM.render(
        <EditProfilePage
          avatar={props.avatar}
          username={props.username}
          email={props.email}
          profile={props.profile}
          csrftoken={csrftoken}
          editEmailUrl={"props.editEmailUrl"}
          />,
        container
    );
}
render();

/*
if (module.hot) {
    console.log("bar");
    module.hot.accept('./edit-profile-page.jsx', () => {
        console.log("HMR!");
        render();
    });
}

*/
