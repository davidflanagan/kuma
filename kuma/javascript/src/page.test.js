//@flow
import React from 'react';
import { create } from 'react-test-renderer';
import DocumentProvider from './document-provider.jsx';
import { fakeDocumentData } from './document-provider.test.js';
import Page from './page.jsx';
import { Titlebar } from './page.jsx';
import UserProvider from './user-provider.jsx';

test('Page snapshot', () => {
    const page = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <Page />
        </DocumentProvider>
    );
    expect(page.toJSON()).toMatchSnapshot();
});

test('Titlebar shows edit and history buttons', () => {
    let titlebar;
    let buttons;

    // No user data; expect no buttons
    titlebar = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <UserProvider.context.Provider value={null}>
                <Titlebar document={fakeDocumentData} />
            </UserProvider.context.Provider>
        </DocumentProvider>
    );
    buttons = titlebar.root.findAll(instance => instance.type === 'button');
    expect(buttons.length).toBe(0);

    // User not logged in, expect no buttons
    titlebar = create(
        <UserProvider.context.Provider value={UserProvider.defaultUserData}>
            <Titlebar document={fakeDocumentData} />
        </UserProvider.context.Provider>
    );
    buttons = titlebar.root.findAll(instance => instance.type === 'button');
    expect(buttons.length).toBe(0);

    // User logged in and contributor, expect two buttons
    titlebar = create(
        <UserProvider.context.Provider
            value={{
                ...UserProvider.defaultUserData,
                isAuthenticated: true,
                isContributor: true
            }}
        >
            <Titlebar
                document={{
                    ...fakeDocumentData,
                    editURL: 'foobar$edit'
                }}
            />
        </UserProvider.context.Provider>
    );
    buttons = titlebar.root.findAll(instance => instance.type === 'button');
    expect(buttons.length).toBe(2);

    // Make the mock window.location property writeable
    // NOTE: This is a little brittle since it assumes that page.jsx
    // is setting window.location and not window.location.href, for example.
    Object.defineProperty(window, 'location', {
        writable: true,
        value: null
    });

    // Clicking first button sets window.location.href to editURL
    // Clicking the second button sets it to the history URL instead
    buttons[0].props.onClick();
    expect(window.location).toBe('foobar$edit');
    buttons[1].props.onClick();
    expect(window.location).toBe('foobar$history');
});
