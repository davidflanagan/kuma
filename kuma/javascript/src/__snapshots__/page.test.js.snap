// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot 1`] = `
Array [
  .emotion-0 {
  height: 5px;
  width: 100%;
  background-image: linear-gradient(-271deg,#206584,#83d0f2);
}

<div
    className="emotion-0"
  />,
  .emotion-39 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  margin: 0 16px;
}

@media (min-width:1024px) {
  .emotion-39 {
    height: 68px;
    grid-template-columns: 221px 5fr minmax(250px,2fr) auto;
    grid-template-areas: "I M S L";
  }
}

@media (min-width:750px) and (max-width:1023px) {
  .emotion-39 {
    height: 122px;
    grid-template-columns: minmax(206px,1fr) 300px auto;
    grid-template-areas: "I S L" "R R R" "M M M";
  }
}

@media (max-width:749px) {
  .emotion-39 {
    font-size: 10px;
    grid-template-columns: repeat(4,1fr);
    grid-template-areas: "I I L L" "S S S S" "M M M M";
  }
}

.emotion-1 {
  display: block;
  grid-area: I;
  width: 200px;
  height: 44px;
  margin-right: 24px;
}

.emotion-0 {
  width: 200px;
  height: 44px;
}

.emotion-2 {
  display: none;
  grid-area: R;
  height: 2px;
  width: 100%;
  background-color: #dce3e5;
}

@media (min-width:750px) and (max-width:1023px) {
  .emotion-2 {
    display: block;
  }
}

.emotion-31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-area: M;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 15px;
  font-weight: bold;
}

.emotion-31 button {
  line-height: 32px;
}

.emotion-9 {
  position: relative;
  pointer-events: auto;
  margin-left: -5px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  padding: 0 5px;
  border: none;
}

.emotion-5:hover {
  background-color: #eee;
}

.emotion-5:focus {
  outline-offset: -3px;
}

.emotion-3 {
  font-size: 75%;
  padding-left: 2px;
}

.emotion-7 {
  position: absolute;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  background-color: white;
  border: solid #83d0f2 1.5px;
  box-shadow: 3px 3px 5px rgba(0,0,0,0.25);
  padding: 10px;
  min-width: 100%;
  display: none;
}

.emotion-7 li {
  padding: 5px;
  white-space: nowrap;
}

.emotion-11 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

.emotion-36 {
  grid-area: S;
  justify-self: stretch;
}

.emotion-35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 8px;
  border: solid 2px #333;
  border-radius: 20px;
  background-color: #fafafa;
  box-sizing: border-box;
  height: 40px;
  min-width: 80px;
  -webkit-flex: 2 1;
  -ms-flex: 2 1;
  flex: 2 1;
}

.emotion-33 {
  fill: #333;
  -webkit-flex: 0 0 20px;
  -ms-flex: 0 0 20px;
  flex: 0 0 20px;
}

.emotion-34 {
  border-width: 0 !important;
  font-size: 14px;
  -webkit-flex: 1 1 100px;
  -ms-flex: 1 1 100px;
  flex: 1 1 100px;
  min-width: 60px;
  color: #000;
  background-color: #fafafa !important;
}

.emotion-38 {
  grid-area: L;
  justify-self: end;
}

.emotion-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1em;
  font-weight: bold;
  color: black;
  margin-left: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-37:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #eee;
}

.emotion-37:focus {
  outline-offset: -3px;
}

<div
    className="emotion-39"
  >
    <a
      className="emotion-1"
      href="/en-US/"
    >
      <svg mock
        alt="MDN Web Docs Logo"
        className="emotion-0"
      />
    </a>
    <div
      className="emotion-2"
    />
    <div
      className="emotion-31 emotion-32"
    >
      <div
        className="emotion-9 emotion-10"
      >
        <button
          className="emotion-5 emotion-6"
          onClick={[Function]}
        >
          Technologies
          <span
            className="emotion-3 emotion-4"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className="dropdown-menu emotion-7 emotion-8"
        >
          <li>
            <a
              href="/en-US/docs/Web/HTML"
            >
              HTML
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/CSS"
            >
              CSS
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/JavaScript"
            >
              JavaScript
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/Guide/Graphics"
            >
              Graphics
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/HTTP"
            >
              HTTP
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/API"
            >
              APIs / DOM
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Mozilla/Add-ons/WebExtensions"
            >
              Browser Extensions
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/MathML"
            >
              MathML
            </a>
          </li>
        </ul>
      </div>
      <div
        className="emotion-11 emotion-12"
        width={16}
      />
      <div
        className="emotion-9 emotion-10"
      >
        <button
          className="emotion-5 emotion-6"
          onClick={[Function]}
        >
          References & Guides
          <span
            className="emotion-3 emotion-4"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className="dropdown-menu emotion-7 emotion-8"
        >
          <li>
            <a
              href="/en-US/docs/Learn"
            >
              Learn web development
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/Tutorials"
            >
              Tutorials
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/Reference"
            >
              References
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/Guide"
            >
              Developer Guides
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web/Accessibility"
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Games"
            >
              Game development
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/Web"
            >
              ...more docs
            </a>
          </li>
        </ul>
      </div>
      <div
        className="emotion-11 emotion-12"
        width={16}
      />
      <div
        className="emotion-9 emotion-10"
      >
        <button
          className="emotion-5 emotion-6"
          onClick={[Function]}
        >
          Feedback
          <span
            className="emotion-3 emotion-4"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className="dropdown-menu emotion-7 emotion-8"
        >
          <li>
            <a
              href="https://support.mozilla.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Get Firefox help
               üåê
            </a>
          </li>
          <li>
            <a
              href="https://stackoverflow.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Get web development help
               üåê
            </a>
          </li>
          <li>
            <a
              href="/en-US/docs/MDN/Community"
            >
              Join the MDN community
            </a>
          </li>
          <li>
            <a
              href="https://github.com/mdn/sprints/issues/new?template=issue-template.md&projects=mdn/sprints/2&labels=user-report&title=test"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report a content problem
               üåê
            </a>
          </li>
          <li>
            <a
              href="https://bugzilla.mozilla.org/form.mdn"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report a bug
               üåê
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="emotion-36"
    >
      <form
        action="/en-US/search"
        className="emotion-35"
        id="nav-main-search"
        method="get"
        role="search"
      >
        <svg mock
          className="emotion-33"
        />
        <input
          className="emotion-34"
          id="main-q"
          name="q"
          placeholder="Search MDN"
          type="search"
        />
      </form>
    </div>
    <div
      className="emotion-38"
    >
      <a
        className="emotion-37"
        data-service="GitHub"
        href="/users/github/login/?next=[fake absolute url]"
        rel="nofollow"
      >
        Sign in
      </a>
    </div>
  </div>,
  .emotion-16 {
  position: relative;
  pointer-events: auto;
  margin-left: -5px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  padding: 0 5px;
  border: none;
}

.emotion-12:hover {
  background-color: #eee;
}

.emotion-12:focus {
  outline-offset: -3px;
}

.emotion-10 {
  font-size: 75%;
  padding-left: 2px;
}

.emotion-25 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 1fr 3fr;
  grid-template-areas: "title title" "crumbs crumbs" "side main";
}

@media (max-width:749px) {
  .emotion-25 {
    grid-template-columns: 1fr;
    grid-template-areas: "title" "crumbs" "main" "side";
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-area: title;
  box-sizing: border-box;
  width: 100%;
  min-height: 106px;
  padding: 12px 24px;
  overflow-x: scroll;
  background-color: #f5f9fa;
  border-bottom: solid 1px #dce3e5;
  border-top: solid 1px #dce3e5;
}

@media (max-width:749px) {
  .emotion-1 {
    min-height: 60px;
    padding: 8px 16px;
  }
}

.emotion-0 {
  font-family: x-locale-heading-primary,zillaslab,"Palatino","Palatino Linotype",x-locale-heading-secondary,serif;
  font-size: 45px;
  font-weight: bold;
}

@media (max-width:749px) {
  .emotion-0 {
    font-size: 28px;
  }
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-area: crumbs;
  box-sizing: border-box;
  width: 100%;
  min-height: 40px;
  padding: 4px 16px 4px 24px;
  border-bottom: solid 1px #dce3e5;
  background-color: #fff;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.emotion-4 {
  display: inline;
  font-size: 14px;
}

.emotion-4a {
  color: #3d7e9a;
}

.emotion-3 {
  font-family: zillaslab,serif;
  font-weight: bold;
  font-size: 18px;
  margin: 0 8px;
}

.emotion-9 {
  height: 20px;
}

.emotion-14 {
  position: absolute;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  background-color: white;
  border: solid #83d0f2 1.5px;
  box-shadow: 3px 3px 5px rgba(0,0,0,0.25);
  padding: 10px;
  min-width: 100%;
  right: 0;
  display: none;
}

.emotion-14 li {
  padding: 5px;
  white-space: nowrap;
}

.emotion-22 {
  grid-area: main;
  box-sizing: border-box;
  width: 100%;
  overflow-x: scroll;
  padding: 30px 24px 30px 12px;
}

@media (max-width:749px) {
  .emotion-22 {
    padding: 15px 12px;
  }
}

.emotion-22 p {
  max-width: 42rem;
}

.emotion-21 {
  margin-top: 32px;
  font-size: 0.88889rem;
  color: #696969;
}

.emotion-21 div {
  margin: 4px 0;
}

.emotion-19 {
  width: 14px;
  height: 14px;
  margin-right: 5px;
  vertical-align: middle;
  fill: #696969;
}

.emotion-20 {
  width: 16px;
  height: 16px;
  margin-right: 5px;
  vertical-align: middle;
  fill: #696969;
}

.emotion-24 {
  grid-area: side;
  box-sizing: border-box;
  width: 100%;
  overflow-x: scroll;
  padding: 30px 12px 30px 24px;
}

@media (max-width:749px) {
  .emotion-24 {
    padding: 15px 12px;
  }
}

.emotion-23 {
  font-size: 20px;
}

<div
    className="emotion-25"
  >
    <div
      className="emotion-1 emotion-2"
    >
      <h1
        className="emotion-0"
      >
        [fake document title]
      </h1>
    </div>
    <div
      className="emotion-18"
    >
      <nav
        className="emotion-8"
        role="navigation"
      >
        <ol>
          <li
            className="emotion-4"
          >
            <a
              href="[fake grandparent url]"
            >
              <span>
                [fake grandparent title]
              </span>
            </a>
            <span
              className="emotion-3"
            >
              ‚Ä∫
            </span>
          </li>
          <li
            className="emotion-4"
          >
            <a
              href="[fake parent url]"
            >
              <span>
                [fake parent title]
              </span>
            </a>
            <span
              className="emotion-3"
            >
              ‚Ä∫
            </span>
          </li>
          <li
            className="emotion-4"
          >
            <span>
              [fake document title]
            </span>
          </li>
        </ol>
      </nav>
      <div
        className="emotion-16 emotion-17"
      >
        <button
          className="emotion-12 emotion-13"
          onClick={[Function]}
        >
          <svg mock
            className="emotion-9"
          />
          <span
            className="emotion-10 emotion-11"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className="dropdown-menu emotion-14 emotion-15"
        >
          <li
            lang="es"
          >
            <a
              href="[fake spanish url]"
              title="Spanish"
            >
              <bdi>
                Espa√±ol
              </bdi>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="text-content emotion-22"
    >
      <article
        dangerouslySetInnerHTML={
          Object {
            "__html": "[fake body HTML]",
          }
        }
        id="wikiArticle"
      />
      <div
        className="emotion-21"
      >
        <div>
          <svg mock
            className="emotion-19"
          />
           
          <strong>
            Contributors to this page:
          </strong>
           
          <span>
            <a
              href="/en-US/profiles/mike"
              rel="nofollow"
            >
              mike
            </a>
          </span>
          <span>
            , 
            <a
              href="/en-US/profiles/ike"
              rel="nofollow"
            >
              ike
            </a>
          </span>
        </div>
        <div>
          <svg mock
            className="emotion-20"
          />
           
          <strong>
            Last updated by:
          </strong>
           
          ike
           
          <time
            dateTime="2019-01-02T03:04:05Z"
          >
            2019-01-02 03:04:05
          </time>
        </div>
      </div>
    </div>
    <div
      className="quick-links emotion-24"
    >
      <div
        className="quick-links-head emotion-23"
      >
        Related Topics
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "[fake quicklinks HTML]",
          }
        }
      />
    </div>
  </div>,
]
`;
